doc/lua_api.md

```diff
@@ -6530,6 +6537,77 @@ Variables:
     * with all functions and userdata values replaced by `true`, calling any
       callbacks here is obviously not possible
 
+Mapgen environment
+------------------
+
+The engine runs the map generator on separate threads, each of these also has
+a Lua environment. Its primary purpose is to allow mods to operate on newly
+generated parts of the map to e.g. generate custom structures.
+Internally it is referred to as "emerge environment".
+
+Refer to the above section for the usual disclaimer on what environment isolation entails.
+
+* `minetest.register_mapgen_dofile(path)`:
+    * Register a path to a Lua file to be imported when a mapgen environment
+      is initialized. You'd put your mapgen-related code in there.
+
+### List of APIs exclusive to the mapgen env
+
+* `minetest.register_on_generated(function(vmanip, minp, maxp, blockseed))`
+    * Called after the engine mapgen finishes a chunk.
+      The chunk data resides in `vmanip`. Other parts of the map are not accessible.
+      The area of the chunk if comprised of `minp` and `maxp`, note that is smaller
+      than the emerged area of the VoxelManip.
+      Note that you don't need to call `read_from_map()` or `write_to_map()` on
+      the VoxelManipulator, doing so is in fact disallowed.
+    * `blockseed`: 64-bit seed number used for this chunk
+* `minetest.save_gen_notify(id, data)`
+    * Saves data for retrieval using the gennotify mechanism (see [Mapgen objects]).
+    * Data is bound to the chunk that is currently being processed, so this function
+      only makes sense inside the above-mentioned generation callback.
```

Comment:
```suggestion
    * Used to transfer data specific to the currently processed chunk across different mapgen envs (see [Mapgen objects]).
    * (execution order?)
 ```
It is yet no clear to me which `register_on_generated` callback is executed first - the one of the main thread or the one inside the mapgen env.  That would need clarification in the introduction text above and perhaps referenced here.

Reply:
Turns out I did not understand the concept of this function. What does it do? How is it helpful in mapgen envs?

Reply:
Does that mean that the separate mapgen envs are executed first so that the gen notifications can be populated? Otherwise I fail to understand how this information could be obtained beforehand.
I also do not see any mention of the execution order.

Reply:
emerge scripting is only initialized after mods load.
at that point the wanted mapgen, flags, noise settings, etc. and also the set of wanted gennotify data is known.

>  I also do not see any mention of the execution order.

You mean in case of multiple `register_mapgen_script` or what else?